application/src/main/java/org/togetherjava/tjbot/commands/free/ChannelMonitor.java

```diff
@@ -281,6 +293,97 @@ public void updateStatusFor(@NotNull Guild guild) {
         return channel;
     }
 
+    /**
+     * Displays the message that will be displayed for users.
+     * <p>
+     * This method detects if any messages have been posted in the channel below the status message.
+     * If that is the case this will delete the existing status message and post another one so that
+     * it's the last message in the channel.
+     * <p>
+     * If it cannot find an existing status message it will create a new one.
+     * <p>
+     * Otherwise it will edit the existing message.
+     *
+     * @param channel the text channel the status message will be posted in.
+     */
+    public void displayStatus(@NotNull TextChannel channel) {
+        final Guild guild = channel.getGuild();
+
+        String messageTxt = buildStatusMessage(guild);
+        MessageEmbed embed = new EmbedBuilder().setTitle(STATUS_TITLE)
+            .setDescription(messageTxt)
+            .setFooter(channel.getJDA().getSelfUser().getName())
+            .setTimestamp(Instant.now())
+            .setColor(MESSAGE_HIGHLIGHT_COLOR)
+            .build();
+
+        Runnable sendStatus = () -> channel.sendMessageEmbeds(embed)
+            .queue(message -> channelIdToMessageIdForStatus.put(channel.getIdLong(),
+                    message.getIdLong()));
+
+        Consumer<Message> editStatus = message -> deleteIfNotLatest(message)
+            .ifPresentOrElse(ignored -> message.editMessageEmbeds(embed).queue(), sendStatus);
+
+        if (!channelIdToMessageIdForStatus.containsKey(channel.getIdLong())) {
+            findExistingStatusMessage(channel).ifPresentOrElse(editStatus, sendStatus);
+        } else {
+            Optional.ofNullable(channelIdToMessageIdForStatus.get(channel.getIdLong()))
+                .map(channel::retrieveMessageById)
+                .ifPresentOrElse(restAction -> restAction.queue(editStatus), sendStatus);
+        }
```

Comment:
This is really bad read, there are too many "branches"
Also besides of the lambda variables it;s getting even harder, take a look into merging the optional's, or something else to improve readability

Reply:
Actually, I did it like this to make sure there's no duplicate code, and also, I got rid of the `complete` that was in the original code, so now this is not blocking at all. I'll try to make it more readable though.

Reply:
6e27d4f

Reply:
You can undo that commit and keep it like it used to be for now, maybe someone else has suggestions

Reply:
you say its worse now? I think it makes the branching way more understandable.